vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNavItem.java

```diff
@@ -180,19 +212,40 @@ private void removeLabelElement() {
      * @see SideNavItem#setPath(Class)
      */
     public void setPath(String path) {
+        setPath(path, RouteParameters.empty());
+    }
+
+    /**
+     * Sets the path in a form or a URL string this navigation item links to.
+     * Note that there is also an alternative way of how to set the link path
+     * via {@link SideNavItem#setPath(Class)}.
+     *
+     * @param path
+     *            The path to link to. Set to null to disable navigation for
+     *            this item.
+     * @param routeParameters
+     *            the route parameters
+     *
+     * @see SideNavItem#setPath(Class)
+     */
+    public void setPath(String path, RouteParameters routeParameters) {
```

Comment:
I thought so as well, but since there is a way to explicitly update the route parameters, I thought that it might at least provide the option to do so with regular string paths. Like you mentioned, it clutters the code with not many practical benefits though. I was on the fence with this, let me remove it.

Reply:
Removed this so that the route parameters are only set along with views.